src/osd/OSD.cc

```diff
@@ -5399,6 +5399,11 @@ bool OSD::_is_healthy()
     if ((float)up < (float)num * cct->_conf->osd_heartbeat_min_healthy_ratio) {
       dout(1) << "is_healthy false -- only " << up << "/" << num << " up peers (less than "
 	      << int(cct->_conf->osd_heartbeat_min_healthy_ratio * 100.0) << "%)" << dendl;
+      if (now - last_mon_heartbeat > cct->_conf->osd_mon_heartbeat_interval) {
+        last_mon_heartbeat = now;
+        dout(1) << "is_healthy false -- checking mon for new map" << dendl;
+        osdmap_subscribe(osdmap->get_epoch() + 1, false);
+      }
```

Comment:
I don't like having side-effects in _is_healthy().  can we instead put this logic in tick() or start_boot() or similar?

Reply:
@liewegas  Updated!